### PR TITLE
[iOS][macOS] Rework top-level file management

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [x] visionOS
- [x] macOS

### Description of changes

High-level summary of changes to structure:

#### Before:
```
.
├── Demo
├── ios
│   ├── FluentUI
│   ├── FluentUI.Demo
│   ├── FluentUI.Resources
│   └── FluentUI.Tests
├── macos
│   ├── FluentUI
│   ├── FluentUIResources-macos
│   ├── FluentUISwiftUITestApp
│   ├── FluentUITestApp
│   ├── FluentUITestViewControllers
│   └── FluentUIUnitTest
├── public
│   └── FluentUI.swift
└── scripts
```

#### After:
```
.
├── Demos
│   ├── FluentUIDemo (SwiftUI WIP)
│   ├── FluentUIDemo_iOS
│   └── FluentUIDemo_macOS
├── Sources
│   ├── FluentUI
│   │   └── FluentUI.swift
│   ├── FluentUI_iOS
│   │   ├── Core
│   │   ├── Components
│   │   └── Resources
│   ├── FluentUI_macOS
│   │   ├── Core
│   │   ├── Components
│   │   └── Resources
│   └── FluentUI_common (name TBD)
├── scripts
└── Tests
    ├── FluentUI_iOS_Tests
    └── FluentUI_macOS_Tests
```

### Binary change

n/a - no changes to build output

### Verification

* Full build and test pass of both libraries
* Built release schemes for both libraries
* **WIP: Podspec validation**

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/2090)

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -2,6 +2,48 @@
...
+            .targetItem(name: "FluentUI_ios", condition: .when(platforms: iOSPlatforms)),
+            .targetItem(name: "FluentUI_macos", condition: .when(platforms: macOSPlatforms))
+        ],
+        path: "Sources/FluentUI"
+    ),
+    .target(
+        name: "FluentUI_ios",
+        path: "Sources/FluentUI_iOS",
+        resources: [
+            .copy("Resources/Version.plist")
```
Comment (joannaquu) [inline Package.swift:21]: do we still need to exclude all the xcfilelists?

**Comment by joannaquu** [comment]
can we also update the readmes. I just went through them and some still point to the xcodeproj oops.

**Comment by joannaquu** [comment]
I believe our localization script also needs updating

**Comment by joannaquu** [comment]
all our doc links will also be broken but maybe for another pr?

File: scripts/xcodebuild_wrapper.sh [hunk 2]
```diff
@@ -52,7 +52,7 @@ function ios_device_build()
 # \param $3+ build commands
 function macos_build()
 {
-    invoke_xcodebuild project "macos/xcode/FluentUI.xcodeproj" "$1" "$2" macosx "${@:3}"
+    invoke_xcodebuild project "Demos/FluentUIDemo_macOS/xcode/FluentUI.xcodeproj" "$1" "$2" macosx "${@:3}"
```
Comment (joannaquu) [inline scripts/xcodebuild_wrapper.sh:55]: are there also plans to remove macOS xcodeproj?

**Comment by mischreiber** [comment]
> Dumb question: Is cocoa pods still supported after this?

@Saadnajmi not dumb! Yes, I made extensive changes to the podspec in commit c3a919f491df9b24e61bbe02f78a0bc07bde8ef3 to keep it building.

File: Package.swift (hunk 1 above)
Comment (mischreiber) [reply Package.swift:21]: Surprisingly no! I'm trying to track down the specific version, but at some point SPM started automatically ignoring them when bundling assets.

**Comment by mischreiber** [comment]
@joannaquu updating docs, loc script, and readme now

File: Package.swift (hunk 1 above)
Comment (joannaquu) [reply Package.swift:21]: ooh sweet!

File: scripts/xcodebuild_wrapper.sh (hunk 2 above)
Comment (joannaquu) [reply scripts/xcodebuild_wrapper.sh:55]: ooh this is for the demo project. ok ignore me!

**Review by joannaquu — APPROVED** [review]

**Pull request merged**